test/powershell/engine/Module/ModuleCmdlet.Tests.ps1

```diff
@@ -0,0 +1,487 @@
+        $mod = Import-Module $modInfo -AsCustomObject
+        $mod.GetType() | Should -BeExactly "PSCustomObject"
+        $mod.'Test-FirstModuleFunction'() | Should -BeExactly "TESTSTRING"
+    }
+
+    It "Re-imports modules from file with new members when -Force is used" {
+        $module = Import-Module $modInfo -PassThru
+
+        $module.ExportedFunctions.Count | Should -Be 1
+        $module.ExportedFunctions.ContainsKey("Test-FirstModuleFunction") | Should -BeTrue
+        $module.ExportedFunctions.ContainsKey("Test-SecondModuleFunction") | Should -BeFalse
+
+        New-Item -Force -Path $modItem.PSPath -Value (((Get-Content $modItem) | Out-String) + "`nfunction Test-SecondModuleFunction { Write-Output 'SECONDSTRING' }")
+
+        $module = Import-Module $modInfo -PassThru -Force
+        Test-SecondModuleFunction | Should -BeExactly "SECONDSTRING"
+    }
+
+    It "Re-imports modules from file with new members when -Force is used with -PassThru" {
+        $module = Import-Module $modInfo -PassThru
+
+        $module.ExportedFunctions.Count | Should -Be 1
+        $module.ExportedFunctions.ContainsKey("Test-FirstModuleFunction") | Should -BeTrue
+        $module.ExportedFunctions.ContainsKey("Test-SecondModuleFunction") | Should -BeFalse
+
+        New-Item -Force -Path $modItem.PSPath -Value (((Get-Content $modItem) | Out-String) + "`nfunction Test-SecondModuleFunction { Write-Output 'SECONDSTRING' }")
+
+        $module = Import-Module $modInfo -PassThru -Force
+        $module.ExportedFunctions.Count | Should -Be 2
+        $module.ExportedFunctions.ContainsKey("Test-FirstModuleFunction") | Should -BeTrue
+        $module.ExportedFunctions.ContainsKey("Test-SecondModuleFunction") | Should -BeTrue
+    }
+
+    It "Re-imports modules from file with new members when -Force is used with -AsCustomObject" {
+        $module = Import-Module $modInfo -PassThru
+
+        $module.ExportedFunctions.Count | Should -Be 1
+        $module.ExportedFunctions.ContainsKey("Test-FirstModuleFunction") | Should -BeTrue
+        $module.ExportedFunctions.ContainsKey("Test-SecondModuleFunction") | Should -BeFalse
+
+        New-Item -Force -Path $modItem.PSPath -Value (((Get-Content $modItem) | Out-String) + "`nfunction Test-SecondModuleFunction { Write-Output 'SECONDSTRING' }")
+
+        $module = Import-Module $modInfo -AsCustomObject -Force
+        $module.'Test-SecondModuleFunction'() | Should -BeExactly "SECONDSTRING"
+    }
+}
+
+Describe "Import-Module with nested modules" -Tag Feature {
+    BeforeAll {
+        $modName = "nestMod"
+        $modDir = "$TestDrive\$modName"
+        $modPath = "$modDir\$modName.psm1"
+        $subModPath = "$modDir\sub.psm1"
+    }
+
+    BeforeEach {
+        New-Item -Path $modDir -ItemType Directory
+    }
+
+    AfterEach {
+        Get-Module $modName | Remove-Module -Force
+        Remove-Item -Path $modDir -Recurse -Force
```

Comment:
Can we exclude BeforeEach and AfterEach and generate new module name and path for every test?